src/YouTube/OAuth.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module YouTube.OAuth
+    ( Port
+    , oauthAuthorizeUrl
+    , oauthCredentials
+    ) where
+
+import           Model (URL)
+import           YouTube.Commons
+import qualified YouTube.Client as YT
+
+import           Control.Concurrent (MVar, forkIO, newEmptyMVar, putMVar, takeMVar)
+import           Control.Monad.IO.Class (liftIO)
+import qualified Data.Text.Lazy as T
+import qualified Network.Wai as Wai
+import qualified Network.Wai.Handler.Warp as Warp
+import           Web.Scotty
```

Comment:
Let's stick to scotty, still better for readability and most scotty dependencies were already there anyway...